pagopa/io-app
# feat: [IOBP-120] Add pagoPA notice decoding to global barcode scan screen

### ⚠️ Depends on https://github.com/pagopa/io-app/pull/4767

## Short description
This PR adds the decoding and handling of pagoPA notice payment in global barcode scan screen.

**Note**: The barcode screen is not currently available inside the app (is not accessible via normal navigation). The below video has been made mocking the navigation from the current barcode scan button in the wallet section.
The screen will be made accessible in **IOBP-66**

https://github.com/pagopa/io-app/assets/6160324/a7ed7140-22c4-4574-b0dd-0222066e3702


## List of changes proposed in this pull request
- Added pagoPA qrcode decoding in `ts/features/barcode/decoders.ts` with tests
- Added pagoPA qrcode scan handling in `ts/features/barcode/screens/BarcodeScanScreen.tsx`
- Added required translation keys
- Refactored `BarcodeFailure` to give more detailed data about the failure
- Added "rest" timer in barcode scan screen to avoid multiple consecutive barcode scan

## How to test
Tests should solve correctly.

To test a pagoPA notice payment in your emulator you can use the `io-dev-server`, then replace any screen with the `BarcodeScanScreen` component.
